#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_clean-doc.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Improve documentation of clean command. Patch suggested by
## DP: Thomas Weber in the mailing list octave-maintainers and applied
## DP: upstream by John W. Eaton, see
## DP: http://www.nabble.com/Implementation-of-clear%28%29-to14497170.html#a14558608

@DPATCH@

--- octave3.0-3.0.0.orig/src/variables.cc
+++ octave3.0-3.0.0/src/variables.cc
@@ -2392,7 +2392,7 @@
 
 DEFCMD (clear, args, ,
   "-*- texinfo -*-\n\
-@deffn {Command} clear [-x] pattern @dots{}\n\
+@deffn {Command} clear [options] pattern @dots{}\n\
 Delete the names matching the given patterns from the symbol table.  The\n\
 pattern may contain the following special characters:\n\
 \n\
@@ -2430,7 +2430,24 @@
 @code{foo} as a function.  Executing @kbd{clear foo} a second time will\n\
 clear the function definition.\n\
 \n\
-With -x, clear the variables that don't match the patterns.\n\
+The following options are available in both long and short form\n\
+@table @code\n\
+@item -all, -a\n\
+Clears all local and global user-defined variables and all functions\n\
+from the symbol table.\n\
+\n\
+@item -exclusive, -x\n\
+Clears the variables that don't match the following pattern.\n\
+\n\
+@item -functions, -f\n\
+Clears the function names and the built-in symbols names.\n\
+@item -global, -g\n\
+Clears the global symbol names.\n\
+@item -variables, -v\n\
+Clears the local variable names.\n\
+@end table\n\
+With the execption of @code{exclusive}, all long options can be used \n\
+without the dash as well.\n\
 @end deffn")
 {
   octave_value_list retval;
